docs/releases/upgrade/0.20.md

```diff
@@ -0,0 +1,421 @@
+# Version 0.20
+
+## Breaking changes
+
+### Change default `join` behavior with regard to null values
+
+Previously, null values in the join key were considered a value like any other value.
+This meant that null values in the left frame would be joined with null values in the right frame.
+This is expensive and does not match default behavior in SQL.
+
+Default behavior has now been changed to ignore null values in the join key.
+The previous behavior can be retained by setting `join_nulls=True`.
+
+**Example**
+
+Before:
+
+```pycon
+>>> df1 = pl.DataFrame({"a": [1, 2, None], "b": [4, 4, 4]})
+>>> df2 = pl.DataFrame({"a": [None, 2, 3], "c": [5, 5, 5]})
+>>> df1.join(df2, on="a", how="inner")
+shape: (2, 3)
+┌──────┬─────┬─────┐
+│ a    ┆ b   ┆ c   │
+│ ---  ┆ --- ┆ --- │
+│ i64  ┆ i64 ┆ i64 │
+╞══════╪═════╪═════╡
+│ null ┆ 4   ┆ 5   │
+│ 2    ┆ 4   ┆ 5   │
+└──────┴─────┴─────┘
+```
+
+After:
+
+```pycon
+>>> df1.join(df2, on="a", how="inner")
+shape: (1, 3)
+┌─────┬─────┬─────┐
+│ a   ┆ b   ┆ c   │
+│ --- ┆ --- ┆ --- │
+│ i64 ┆ i64 ┆ i64 │
+╞═════╪═════╪═════╡
+│ 2   ┆ 4   ┆ 5   │
+└─────┴─────┴─────┘
+>>> df1.join(df2, on="a", how="inner", join_nulls=True)  # Keeps previous behavior
+shape: (2, 3)
+┌──────┬─────┬─────┐
+│ a    ┆ b   ┆ c   │
+│ ---  ┆ --- ┆ --- │
+│ i64  ┆ i64 ┆ i64 │
+╞══════╪═════╪═════╡
+│ null ┆ 4   ┆ 5   │
+│ 2    ┆ 4   ┆ 5   │
+└──────┴─────┴─────┘
+```
+
+### Preserve left and right join keys in outer joins
+
+Previously, the result of an outer join did not contain the join keys of the left and right frames.
+Rather, it contained a coalesced version of the left key and right key.
+This loses information and does not conform to SQL behavior.
+
+The behavior has been changed to include the original join keys.
+Name clashes are solved by appending `_right` to the right join key name.
```

Comment:
```suggestion
Name clashes are solved by appending `suffix` ("_right" by default) to the right join key name.
```

Reply:
Thanks, will update. I forgot we had that parameter.